SYCL/InvokeSimd/ESIMD_to_SPMD_error_call_vadd_1d.cpp

```diff
@@ -0,0 +1,218 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// Check that full compilation works:
+// RUN: not %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out 2>&1 | FileCheck %s
+//
+// Tests invoke_simd support in the compiler/headers
+
+/*
+  Test case specification: Test and report errors if invoked ESIMD function
+  calls SPMD function.
+
+  * In this test case, ESIMD_CALLEE_doVadd() calls SPMD_CALLEE_doVadd().
+  * Currently, the compilation fails, which is good.
+
+  From Konst:
+
+  "Today we don’t support calling SYCL_EXTERNAL SPMD functions from ESIMD code,
+  as semantics is not fully clear and will require specification.
+  SYCL_EXTERNAL function is SPMD if it is not marked with SYCL_ESIMD_FUNCTION.
+  W/o SYCL_EXTERNAL, a function w/o SYCL_ESIMD_FUNCTION is treated as ESIMD if
+  called from within ESIMD code (callgraph is built). The same function can be
+  called from SYCL, in which case there will be two copies created and compiled
+  differently (via scalar and vector BEs)."
```

Comment:
Suggest to remove this. There are lots of things ESIMD spec does not specify, and which are kind of assumed. 
```suggestion
This test case is based on the assumption that ESIMD functions can't call SPMD functions, as there is no mechanism for this described in ESIMD specification.
```